omnigibson/utils/usd_utils.py

```diff
@@ -925,3 +1377,97 @@ def get_world_prim():
         Usd.Prim: Active world prim in the current stage
     """
     return lazy.omni.isaac.core.utils.prims.get_prim_at_path("/World")
+
+
+def scene_relative_prim_path_to_absolute(scene, relative_prim_path):
+    """
+    Converts a scene-relative prim path to an absolute prim path.
+
+    Args:
+        scene (Scene): Scene object that the prim is in. None if it's global.
+        relative_prim_path (str): Relative prim path in the scene
+
+    Returns:
+        str: Absolute prim path in the stage
+    """
+    # Special case for OmniGraph prims
+    if relative_prim_path.startswith("/OmniGraph"):
+        return relative_prim_path
+
+    # Make sure the relative path is actually relative
+    assert not relative_prim_path.startswith("/World"), f"Expected relative prim path, got {relative_prim_path}"
+
+    # When the scene is set to None, this prim is not in a scene but is global e.g. like the
+    # viewer camera or one of the scene prims.
+    if scene is None:
+        return "/World" + relative_prim_path
+
+    return scene.prim_path + relative_prim_path
+
+
+def absolute_prim_path_to_scene_relative(scene, absolute_prim_path):
+    """
+    Converts an absolute prim path to a scene-relative prim path.
+
+    Args:
+        scene (Scene): Scene object that the prim is in. None if it's global.
+        absolute_prim_path (str): Absolute prim path in the stage
+
+    Returns:
+        str: Relative prim path in the scene
+    """
+    # Special case for OmniGraph prims
+    if absolute_prim_path.startswith("/OmniGraph"):
+        return absolute_prim_path
+
+    assert absolute_prim_path.startswith("/World"), f"Expected absolute prim path, got {absolute_prim_path}"
+
+    # When the scene is set to None, this prim is not in a scene but is global e.g. like the
+    # viewer camera or one of the scene prims.
+    if scene is None:
+        assert not absolute_prim_path.startswith(
+            "/World/scene_"
+        ), f"Expected global prim path, got {absolute_prim_path}"
+        return absolute_prim_path[len("/World") :]
+
+    return absolute_prim_path[len(scene.prim_path) :]
+
+
+def deep_copy_prim(source_root_prim, dest_stage, dest_root_path):
```

Comment:
this is cool! what is this used for?

Reply:
It was supposed to be used for creating a full scene USD file during prebuild. Does not work great sadly - the partial files get left open. Will be fixed in a follow up